### PR TITLE
[no-release-notes] progress on Diff and Merge

### PR DESCRIPTION
**Pull request by druvv**
Unkips TestDiffTableFunction.

- Fixes typo in Diff that caused panics for some of the tests
- Conflicts can now be resolved through SQL
- Merge stats for a FF is now being calculated. Still need to calculate merge stats for a pure merge. Merge stats is largely untested by engine tests, leaving this as a TODO for now.

Continuing to work on Diff. Ran into a difference in behavior between the old format and new format. Alter table currently preserves the column tag during a rename. Because the new format purely matches columns based on name, we need to have alter table change the column tag in the old format.

### CODE REVIEW

File: go/libraries/doltcore/diff/diff_summary.go [hunk 1]
```diff
@@ -33,20 +37,43 @@ type DiffSummaryProgress struct {
...
 
-// todo: make package private once dolthub is migrated
-// Summary reports a summary of diff changes between two values
 // Summary reports a summary of diff changes between two values
-func Summary(ctx context.Context, ch chan DiffSummaryProgress, from, to types.Map) (err error) {
+// todo: make package private once dolthub is migrated
+func Summary(ctx context.Context, ch chan DiffSummaryProgress, from, to durable.Index, fromSch, toSch schema.Schema) (err error) {
+	ch <- DiffSummaryProgress{OldSize: from.Count(), NewSize: to.Count()}
+
+	if from.Format() == types.Format_DOLT_1 {
```
Comment (zachmu) [inline go/libraries/doltcore/diff/diff_summary.go:45]: Extract a method for each format

File: go/libraries/doltcore/diff/diff_summary.go [hunk 2]
```diff
@@ -195,3 +274,45 @@ func reportKeylessChanges(ctx context.Context, change *diff.Difference, ch chan<
...
 }
+
+// MapSchemaBasedOnName can be used to map column values from one schema to
+// another schema. A column in |inSch| is mapped to |outSch| if they share the
+// same name and primary key membership status. It returns ordinal mappings that
+// can be use to map key, value val.Tuple's of schema |inSch| to a sql.Row of
+// |outSch|. The first ordinal map is for keys, and the second is for values. If
+// a column of |inSch| is missing in |outSch| then that column's index in the
+// ordinal map holds -1.
+func MapSchemaBasedOnName(inSch, outSch schema.Schema) (val.OrdinalMapping, val.OrdinalMapping, error) {
```
Comment (zachmu) [inline go/libraries/doltcore/diff/diff_summary.go:285]: Seems like the kind of thing that should have a unit test with a bunch of interesting test cases

File: go/libraries/doltcore/sqle/dtables/conflicts_tables.go [hunk 3]
```diff
@@ -51,12 +60,34 @@ func NewConflictsTable(ctx *sql.Context, tblName string, root *doltdb.RootValue,
...
+		if err != nil {
+			return nil, err
+		}
+		schs.Schema, schs.MergeSchema = schs.Base, schs.Base
+	}
 
-	sqlSch, err := sqlutil.FromDoltSchema(doltdb.DoltConfTablePrefix+tblName, rd.GetSchema())
+	var rd *merge.ConflictReader
+	var confSch schema.Schema
+	if tbl.Format() == types.Format_DOLT_1 {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dtables/conflicts_tables.go:77]: Should extract a `getConflictSchema` for this initialization logic

File: go/libraries/doltcore/sqle/dtables/conflicts_tables.go [hunk 4]
```diff
@@ -51,12 +60,34 @@ func NewConflictsTable(ctx *sql.Context, tblName string, root *doltdb.RootValue,
...
+	if schs.Base == nil || schs.Schema == nil || schs.MergeSchema == nil {
+		schs.Base, err = tbl.GetSchema(ctx)
+		if err != nil {
+			return nil, err
+		}
+		schs.Schema, schs.MergeSchema = schs.Base, schs.Base
+	}
 
-	sqlSch, err := sqlutil.FromDoltSchema(doltdb.DoltConfTablePrefix+tblName, rd.GetSchema())
+	var rd *merge.ConflictReader
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dtables/conflicts_tables.go:75]: Not needed outside the else block

File: go/libraries/doltcore/sqle/dtables/conflicts_tables.go [hunk 5]
```diff
@@ -93,13 +126,195 @@ func (ct ConflictsTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error)
...
+			return nil, err
+		}
+		r[itr.t+itr.kd.Count()+i] = f
+	}
+
+	return r, nil
+}
+
+func (itr prollyConflictRowIter) Close(ctx *sql.Context) error {
+	return nil
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dtables/conflicts_tables.go:231]: Conflict iter doesn't need closing?

File: go/libraries/doltcore/sqle/dtables/conflicts_tables.go [hunk 6]
```diff
@@ -93,13 +126,195 @@ func (ct ConflictsTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error)
...
+			return err
+		}
+	}
+
+	updatedRoot, err := cd.ct.root.PutTable(ctx, cd.ct.tblName, updatedTbl)
+	if err != nil {
+		return err
+	}
+
+	return cd.ct.rs.SetRoot(ctx, updatedRoot)
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dtables/conflicts_tables.go:317]: This doesn't seem right... in particular, you can delete from more than one table in a single statement. We really need root updates to be restricted to write sessions, not individual table editors. For now just add a TODO

**Review by zachmu — APPROVED** [review]
LGTM, just a few comments. You can address them in this PR or in a followup, your call

File: go/libraries/doltcore/sqle/dtables/conflicts_tables.go (hunk 5 above)
Comment (druvv) [reply go/libraries/doltcore/sqle/dtables/conflicts_tables.go:231]: nope

File: go/libraries/doltcore/diff/diff_summary.go (hunk 2 above)
Comment (druvv) [reply go/libraries/doltcore/diff/diff_summary.go:285]: Added a TODO for this.

**Pull request merged**